### PR TITLE
feat: add brick generator

### PR DESCRIPTION
**Pull request by renancaraujo**
## Description

feat: generate the brick

Generate the brick on top of the app base.

Blocked by #3

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: brick/hooks/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: very_good_wear_app_hooks
+
+environment:
+  sdk: ">=2.17.0 <3.0.0"
```
Comment (omartinma) [inline brick/hooks/pubspec.yaml:4]: on the previous PR the minimum is 2.18, is ok to have different versions?

Comment (alestiago) [reply brick/hooks/pubspec.yaml:4]: I think it should be bumped to 2.18.0

File: tool/generator/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: generator
+description: A template generator for Very Good CLI.
+publish_to: none
+
+environment:
+  sdk: ">=2.13.0-0 <3.0.0"
```
Comment (alestiago) [inline tool/generator/pubspec.yaml:6]: Can we bumped the SDK of the generator?

**Review by alestiago — APPROVED** [review]

File: tool/generator/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: generator
+description: A template generator for Very Good CLI.
```
[the remark this answers is not in the document]
Comment (renancaraujo) [reply tool/generator/pubspec.yaml:2]: removed the description since this is not that important

**Review by alestiago — APPROVED** [review]
The generator code feels hacky (as it follows the other generator patterns), I think we should invest some time into making them boring at some point, but that's beyond this PR.

**Pull request merged**